paimon-core/src/main/java/org/apache/paimon/tag/TagTimeExpire.java

```diff
@@ -52,24 +59,41 @@ private TagTimeExpire(
         this.callbacks = callbacks;
     }
 
-    public void run() {
+    public List<String> expire() {
         List<Pair<Tag, String>> tags = tagManager.tagObjects();
+        FileIO fileIO = snapshotManager.fileIO();
```

Comment:
inline this to line 73:
tagFileStatus = snapshotManager.fileIO().getFileStatus(tagManager.tagPath(tagName));